### PR TITLE
displayio: Fix several bugs (transparency and palettes of OnDiskBitmaps)

### LINKED ISSUE #3723: BMP image loading and rendering issues
Certain BMP varieties fail to load or display as intended. Behavior varies depending on image type, size, and whether using displayio.OnDiskBitmap vs adafruit_imageload library. (Last image is skipped because it's known that imageload doesn't handle 24-bit BMPs -- this is NOT a bug report or feature request for that -- the other images represent the issues being raised here.)

Test platform is Adafruit MagTag running adafruit-circuitpython-adafruit_magtag_2.9_grayscale-en_US-20201117-a0d3050.uf2. Test code and data attached. Comments in code describe required libraries and walk through the issues.

[circuitpy-image-test.zip](https://github.com/adafruit/circuitpython/files/5564267/circuitpy-image-test.zip)

Elsewhere I mentioned an issue where 4-bit pixels might be extracted in the wrong hi/lo order. I'm not seeing that here now, so please disregard that. Will add comments/changes here if I'm able to reproduce it again.

### PR DESCRIPTION
**Pull request by jepler**
The transparent_color field was never initialized.  I _think_ this means its value was always set to 0, or the blackest of blacks.  Instead, initialize it to the sentinel value, newly given the name NO_TRANSPARENT_COLOR.

This exposed a second problem: The test for whether there was an existing transparent color was wrong (backwards).  I am guessing that this was not found due to the first bug; since the converter had a transparent color, the correct test would have led to always getting the error "Only one color can be transparent at a time".

Closes #3723

### CODE REVIEW

**Comment by jepler** [comment]
@PaintYourDragon Test the artifacts from this build for the transparency/full black problem.  The "testpattern4.bmp" file _DOES_ still seem to crash, and I'll address it in a separate commit (probably a separate PR)

**Comment by jepler** [comment]
This PR has been updated with a fix for the 4-bit bitmap.  

[Microsoft documentation](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader) says:
    
> If biCompression equals BI_RGB and the bitmap uses 8 bpp or less, the bitmap has a color table immediatelly following the BITMAPINFOHEADER structure. The color table consists of an array of RGBQUAD values. The size of the array is given by the biClrUsed member. If biClrUsed is zero, the array contains the maximum number of colors for the given bitdepth; that is, 2^biBitCount colors.

We were treating 0 colors as "no palette" during construction of the OnDiskBitmap, and then indexing into the palette anyway during common_hal_displayio_ondiskbitmap_get_pixel.  This caused unpredictable results—my pygamer showed the image in black and blue, and my magtag crashed.

**Comment by jepler** [comment]
[CI failure is a problem uploading artifacts, which is reported at https://github.com/actions/upload-artifact/issues/116 but not metoo'd by us yet.  should not affect accepting this PR]

**Comment by PaintYourDragon** [comment]
With latest from S3…
adafruit-circuitpython-adafruit_magtag_2.9_grayscale-en_US-20201217-8f9cd70.uf2
(Adafruit CircuitPython 6.1.0-beta.2-104-g8f9cd7075 in REPL)

displayio still exhibiting transparency issue at various depths and 4bpp image crash. Is there some still-pending build in the queue I should be waiting for?

**Comment by jepler** [comment]
@PaintYourDragon Until this PR is merged, it won't appear on S3.  You need to get it from the Actions build instead.

How to navigate there: Click the "Checks" tab below the PR title, then click "artifacts", and find the "adafruit-magtag-..." zip near the top of the list.

Direct-er link: https://github.com/adafruit/circuitpython/suites/1687216126/artifacts/31633147

**Comment by dhalbert** [comment]
I'm happy to approve this after @PaintYourDragon verifies the fix.

**Comment by PaintYourDragon** [comment]
The “artifact” version works and appears to fix both the transparency and crash-on-odd-size-BMP issues. Thanks.

**Review by tannewt — APPROVED** [review]
Thanks @jepler

**Pull request merged**
